addons/crm/models/res_partner.py

```diff
@@ -63,3 +63,15 @@ def action_view_opportunity(self):
         else:
             action['domain'] = [('partner_id.id', '=', self.id)]
         return action
+
+    def _fields_sync(self, values):
+        # sync salesteam with parent
+        if not self.team_id and not values.get('team_id') and values.get('parent_id') and self.company_type == 'person':
+            return super().write({'team_id': self.parent_id.team_id.id})
+        return super()._fields_sync(values)
+
+    @api.onchange('parent_id')
```

Comment:
the onchange should be just after the computed method
https://www.odoo.com/documentation/15.0/developer/misc/other/guidelines.html#symbols-and-conventions